### ISSUE TITLE
PixelCopy is taking screenshots of layers outside of the composable

### ISSUE THREAD

**Issue by nubpro** [PatilShreyas/Capturable#101]
I was studying how the inners of this library and discovered there's a slight problem with the `drawBitmapWithPixelCopy()` method which is used to capture the `View` into bitmap.

You can test this by modifying the library to force use [`drawBitmapWithPixelCopy()` over `view.drawToBitmap()`](https://github.com/PatilShreyas/Capturable/blob/3cb15ba2917e21b4f90fe78de1770e66a0b4754c/capturable/src/main/java/dev/shreyaspatil/capturable/Capturable.kt#L125-L153).
1. Add `throw java.lang.IllegalArgumentException("force throw")` before [this line](https://github.com/PatilShreyas/Capturable/blob/3cb15ba2917e21b4f90fe78de1770e66a0b4754c/capturable/src/main/java/dev/shreyaspatil/capturable/Capturable.kt#L130) and use any SDK Android Oreo or above.

2. Copy and use the composable below.
```
@Composable
fun Main() {
    Box {
        val captureController = rememberCaptureController()
        val context = LocalContext.current

        Capturable(controller = captureController, onCaptured = { bitmap, error ->
            bitmap?.let {
                File(context.filesDir, "screenshot.png")
                    .writeBitmap(bitmap.asAndroidBitmap(), Bitmap.CompressFormat.PNG, 85)
            }
            Log.i("capturable", context.filesDir.toString())

            if (error != null) {
                Log.e("ERROR!", error.toString())
            }
        }) {
            Image(painter = painterResource(id = R.drawable.poster),
                contentDescription = null,
                modifier = Modifier.fillMaxWidth(),
                contentScale = ContentScale.Crop)
        }


        Button(onClick = {
            captureController.capture()
        }) {
            Text("Take a screenshot!!!!")
        }

        Text(text = "You shouldn't see me", fontSize = 140.sp, color= Color.White, modifier = Modifier.padding(15.dp))
    }
}
```

3. Hit the screen button.
4. Locate the screenshot.
5. You will see that Capturable took a screenshot which contains the forbidden text, even though the text is not within the Capturable composable.

![image](https://github.com/PatilShreyas/Capturable/assets/762914/09eea35e-dc82-4a1a-a5f8-8c27713d296f)

Any help is very much appreciated by our team!

**Comment by abinhho** [comment]
Same here.

**Comment by abinhho** [comment]
Related: https://stackoverflow.com/a/60360589/9585130

**Comment by nikoloz14** [comment]
Any news?

**Comment by PatilShreyas** [comment]
This issue has been fixed and released in version 2.0.0.

**Comment by nubpro** [comment]
> This issue has been fixed and released in version 2.0.0.

Amazing work. Bravo!
